### ISSUE TITLE
Pages and blog posts not showing up on sitemap

### ISSUE THREAD

**Issue by DjMorgul** [semperfiwebdesign/all-in-one-seo-pack#1947]
**Describe the bug**
Pages and blog posts aren't showing up on sitemap, none of them.

**To Reproduce**
Even on my settings I have both the checkboxes for pages and posts selected, and no category selected for exclusion, none of the pages nor the blog posts show up on the generated sitemap.

**Expected behavior**
All published pages and blog posts should appear on the sitemap.

**Version Info (please complete the following information):**
 - WordPress version: 4.9.8
 - All in One SEO Pack version: 2.8
 - PHP version: 7.1.22

**Additional context**
I have additional pages that I added manually, for closed bbPress forum posts, because they weren't showing up on the sitemap either #1620

**Comment by michaeltorbert** [comment]
You can submit your login information to our secure form here and we'll take a look: https://semperplugins.com/contact/submit-login-information/

Make sure to also provide the URLs for at least one page and one post that aren't showing up.

**Comment by arnaudbroes** [comment]
@DjMorgul thank you for sending us an administrator login. I accessed your website and I immediately was able to locate both of the example URLs that you mentioned. 


![image](https://user-images.githubusercontent.com/18156697/46048483-dec3e480-c129-11e8-9461-27679e1ad55f.png)

**Comment by DjMorgul** [comment]
Proof that Google is also getting a sitemap with only 23 URLs.

![2018-09-29-152554_1275x1051_scrot](https://user-images.githubusercontent.com/14842546/46246330-36996e80-c3fc-11e8-8593-b5ff06e347c4.png)

**Comment by arnaudbroes** [comment]
@DjMorgul apparently, your sitemap is showing 53 URLs when logged in as an admin and only 23 when not logged in as an adminstrator user. We're going to troubleshoot the issue and let you know more once we've identified what's causing this behaviour.

**Comment by arnaudbroes** [comment]
I also found out that playing around with the settings or deactivating AIOSEOP has no effect on the sitemap at all. It's still there. @michaeltorbert can you take a look at this?

**Comment by DjMorgul** [comment]
That's curious. Just in case, I checked that there's no file named sitemap.xml (nor anything similar) on the root folder of my webpage.

**Comment by arnaudbroes** [comment]
@DjMorgul I did too. ;-)

**Comment by arnaudbroes** [comment]
@wpsmort I'll send an email with the credentials and link to the test site.

**Comment by wpsmort** [comment]
@arnaudbroes @DjMorgul This does not appear to be a bug with AIOSEOP.  It appears to be a problem with bbPress which is not installed correctly or is corrupt.  When bbPress is installed you should see menu items for Forums, Topics and Replies and a sub menu item under Settings for Forum Settings.  See screenshot below. These are not appearing on the clone site and don't appear if you reinstall bbPress.  Our XML Sitemap works fine until the point that you add bbPress Topics as a post type in the XML Sitemap settings.  

The reason for logged in users getting a number of URLs to someone not logged in was a caching issue.  Clearing the Varnish cache on the clone site resolved this.

**Comment by DjMorgul** [comment]
Please, let me know when I can stop allowing you that admin access. Take into account that it's my production site.

**Comment by michaeltorbert** [comment]
@DjMorgul You can remove our access, as it looks like @wpsmort has responded and closed the issue.

**Comment by DjMorgul** [comment]
But I can see those bbPress menu entries, without touching a thing...

![2018-10-18-154326_1362x1057_scrot](https://user-images.githubusercontent.com/14842546/47158934-02bfb380-d2ed-11e8-93e9-5f6a141be767.png)

And I'm not using any caching plugin right now.

**Comment by DjMorgul** [comment]
Okay, I think you weren't seeing those menu entries because the user I provided for you didn't have the bbPress role for that. Now I have changed its role from Participant to Keymaster.

**Comment by wpsmort** [comment]
@DjMorgul I tested thoroughly on a clone of your site on our private development server and the only way I could get this to work was to delete the settings for bbPress and reinstall that plugin.  As soon as I did that the sitemap showed the correct number of URLs.  I don't see which setting in bbPress could be causing this but it definitely looks to be a bbPress issue.

**Comment by DjMorgul** [comment]
I confirm it was a bbPress issue. I identified where the problem was, and reported it to their bug tracker: https://bbpress.trac.wordpress.org/ticket/3234

Now I just created a patch to fix the issue, which works finely on my site, and will prevent bbPress from messing up with foreign queries in such a way (with their code to exclude private forums and their topics and replies). It's attached to that issue on bbPress tracker.

Thank you again for your help on solving this, and for pointing me in the right direction to investigate it.
